### PR TITLE
Driven sampler

### PR DESCRIPTION
**Pull request by drckf**
This branch implements a novel sequential Monte Carlo sampler that is driven out of equilibrium by an auxiliary variable multiplying the weights of the model. The auxiliary variable (beta) fluctuates about a mean of 1, so that the sampling distribution is -- on average -- close to the model distribution. But, the mixing rate of the Markov chain is increased quite a bit. This results in much better training without adding any additional computational overhead. The performance difference between SequentialMC and DrivenSequentialMC is especially pronounced for the GaussianRBM, which pretty much fails to learn a good generative model without the non-equilibrium driver.

I've also tried to improve the modularity of samplers and optimizers so that the user has more flexibility in specifying them.

### CODE REVIEW

**Comment by jrwalsh1** [comment]
there are a couple of `pyc` files in here - should remove them and add `*.pyc` to `.gitignore`.

File: paysage/optimizers.py [hunk 1]
```diff
@@ -2,17 +2,55 @@
...
+
+
+class PowerLawDecay(Scheduler):
+
+    def __init__(self, lr_decay=0.1):
+        super().__init__()
+        self.lr_decay = lr_decay
+
+    def get_lr(self):
+        self.lr = 1 / (1 + self.lr_decay * self.epoch)
```
Comment (jrwalsh1) [inline paysage/optimizers.py]: did you want `self.lr_decay` in the numerator here?  Otherwise you're always starting from a learning rate of 1.

Comment (drckf) [inline paysage/optimizers.py]: In the optimizers, I multiply the gradient by `lr * stepsize`. This separates out the initial learning rate from the decay.

**Comment by drckf** [comment]
Fixed the .pyc issue. Just needed to remove the git cache.

**Comment by jrwalsh1** [comment]
OK, great.  I tried adding them again and `.gitignore` caught them; they must have just predated the ignore update.

**Comment by jrwalsh1** [comment]
Awesome addition!
